src/rospkg/rospack.py

```diff
@@ -391,6 +392,36 @@ def stack_of(self, package):
             else:
                 d = os.path.dirname(d)
 
+    def get_licenses(self, pkg_name, implicit=True):
+        """
+        @summary: Return a list of licenses and the packages in the dependency tree
+            for the given package. Special value 'ERR' is used as the license for the
+            packages that license was not detected for.
+        @param pkg_name: Name of the package the dependency tree begins from.
+        @return OrderedDict of license name and a list of packages.
+        @rtype { k, [d] }
+        @raise ResourceNotFound
+        """
+        license_dict = defaultdict(list)
+
+        self.get_depends(name=pkg_name, implicit=implicit)
+
+        for p_name, manifest in self._manifests.items():
+            for license in manifest.licenses:
+                license_dict[license].append(p_name)
+
+        # Traverse for Non-ROS, system packages
+        pkgnames_rosdep = self.get_rosdeps(package=pkg_name, implicit=implicit)
+        for pkgname_rosdep in pkgnames_rosdep:
+            license_dict[None].append(pkgname_rosdep)
```

Comment:
So to answer from a different perspective, comparing `get_depends` (already a part of `rospkg` and `get_licenses` method that is being added in this PR,
- `get_depends` returns a list of `name` elements in the manifest.
- `get_licenses` returns a list of `name` elements grouped by `license` in the manifest.

Because both only return elements in the manifest (`package.xml`) without adding any info external from the manifest file. So it makes sense to me to add `get_licenses` capability to `rospkg`.